sdv/single_table/base.py

```diff
@@ -112,7 +112,9 @@ def __init__(
     ):
         self._validate_inputs(enforce_min_max_values, enforce_rounding)
         self.metadata = metadata
+        self.table_name = Metadata.DEFAULT_SINGLE_TABLE_NAME
         if isinstance(metadata, Metadata):
+            self.table_name = metadata._get_single_table_name()
```

Comment:
I'm confused about this scenario. If multiple tables are in the metadata, then we error in the single table case so we don't need the warning. Maybe we should also just error if the user doesn't pass a table name in instead of replacing with a default

Reply:
This scenario is if there is a single table, we want to make sure `get_metadata` returns the single table object with the correct name attached to the table (same one that is passed in) if there isn't a name (e.g. legacy `SingleTableMetadata` passed in) we can use the default name.

Reply:
If it's a legacy metadata then it won't enter this branch of the if right? And if it's the legacy metadata why do we need a name?

Reply:
You're right, legacy metadata will just have the default name so it's not the best example.

When we call `get_metadata` that comes out with the class `Metadata`. That table should match whatever was put into the class so the user can refer to the table using the same table name.